src/Native/Runtime/GCHelpers.h

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+//
+// Unmanaged GC helpers
+//
+
+#ifdef WRITE_BARRIER_CHECK
+extern uint8_t* g_GCShadow;
+extern uint8_t* g_GCShadowEnd;
+extern "C" uint8_t* g_lowest_address;
+extern "C" uint8_t* g_highest_address;
+#endif
+
+extern "C" uint32_t* g_card_table;
+static UInt32 INVALIDGCVALUE = 0xcccccccd;
+
+FORCEINLINE void RhpBulkWriteBarrier(void* pMemStart, UInt32 cbMemSize)
```

Comment:
It is wasteful to FORCEINLINE this in all place where it is used (e.g. the use in threadstore.cpp) and it collides with the assembly RhpBulkWriteBarrier for !CORERT. We should give it a different name - without Rhp prefix. Maybe InlinedBulkWriteBarrier ?